### PR TITLE
[NB-4861] Port recent Notebooks updates including memory usage reporting improvement and DR library version upgrades

### PR DESCRIPTION
**Pull request by c-h-russell-walker**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary

Port recent Notebooks updates:

- memory usage reporting improvement 
  - https://github.com/datarobot/notebooks/pull/2854
- DR library version upgrades
  - https://github.com/datarobot/notebooks/pull/2855

### CODE REVIEW

**Comment by c-h-russell-walker** [comment]
These changes have now made it to EU and US Prod. for the built-in environments notebooks uses and they are working as expected.

**Review by tpadr — APPROVED** [review]
https://datarobot.atlassian.net/browse/NB-3655 tested and safe to merge

**Review by dmytrokarpovych — APPROVED** [review]
thank you 🙇

File: public_dropin_notebook_environments/python38_snowflake_notebook/agent/cgroup_watchers.py [hunk 1]
```diff
@@ -51,7 +51,13 @@ def __init__(self) -> None:
         self._cpuacct_usage_file = cgroup_cpuacct_dir / "cpuacct.usage"
 
     def memory_usage_in_bytes(self) -> int:
-        return self._read_metric(self._memory_usage_file)
+        memory_stat_str = self._memory_usage_file.read_text()
+        total_rss_str = next(
+            iter([stat for stat in memory_stat_str.split("\n") if stat.startswith("total_rss")]),
+            "0",
+        )
+        total_rss = int(total_rss_str.split(" ")[-1])
```
Comment (zohar-mizrahi) [inline public_dropin_notebook_environments/python38_snowflake_notebook/agent/cgroup_watchers.py:59]: nit: I'd recommend using `split` without arguments, thus it splits the string at whitespace characters (spaces, tabs, etc).
Same for all there corresponding places in this PR.

**Review by zohar-mizrahi — APPROVED** [review]

File: public_dropin_notebook_environments/python38_snowflake_notebook/agent/cgroup_watchers.py (hunk 1 above)
Comment (c-h-russell-walker) [reply public_dropin_notebook_environments/python38_snowflake_notebook/agent/cgroup_watchers.py:59]: Thanks for taking a look - I'll defer to @tpadr in order for us to keep it as close as possible to the existing code in our notebooks repo - code seen here:

https://github.com/datarobot/notebooks/pull/2854

**Comment by c-h-russell-walker** [comment]
Confirmed these changes from this PR have been built and released to Prod. - seen here:

Screenshot 2023-11-29 at 3 45 23 PM

**Pull request merged**